Buildings/Experimental/OpenBuildingControl/CDL/SetPoints/CombiTimeTable.mo

```diff
@@ -0,0 +1,524 @@
+within Buildings.Experimental.OpenBuildingControl.CDL.SetPoints;
+block CombiTimeTable
+  "Table look-up with respect to time and linear/periodic extrapolation methods (data from matrix/file)"
+  extends Modelica.Blocks.Icons.Block;
+  parameter Boolean tableOnFile=false
+    "= true, if table is defined on file or in function usertab"
+    annotation (Dialog(group="Table data definition"));
+  parameter Real table[:, :] = fill(0.0, 0, 2)
+    "Table matrix (time = first column; e.g., table=[0,2])"
+    annotation (Dialog(group="Table data definition",enable=not tableOnFile));
+  parameter String tableName="NoName"
+    "Table name on file or in function usertab (see docu)"
+    annotation (Dialog(group="Table data definition",enable=tableOnFile));
+  parameter String fileName="NoName" "File where matrix is stored"
+    annotation (Dialog(
+      group="Table data definition",
+      enable=tableOnFile,
+      loadSelector(filter="Text files (*.txt);;MATLAB MAT-files (*.mat)",
+          caption="Open file in which table is present")));
+  parameter Boolean verboseRead=true
+    "= true, if info message that file is loading is to be printed"
+    annotation (Dialog(group="Table data definition",enable=tableOnFile));
+  parameter Integer columns[:]=2:size(table, 2)
+    "Columns of table to be interpolated"
+    annotation (Dialog(group="Table data interpretation"));
+  parameter Modelica.Blocks.Types.Smoothness smoothness=Modelica.Blocks.Types.Smoothness.LinearSegments
```

Comment:
At some point we've had a discussion with a conclusion to retain any types from Modelica library for now. There are a bunch of such types at other places, e.g. ```Modelica.SIunits.Time```, several others in Psychrometrics. Let me know if you'd like to reconsider that.